internal/rbd/group_controllerserver.go

```diff
@@ -130,6 +132,24 @@ func (cs *ControllerServer) CreateVolumeGroupSnapshot(
 			"failed to get existing one with name %q: %v", vgsName, err)
 	}
 
+	creds, err := mgr.GetCredentials()
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	errList := make([]error, 0)
+	for _, volume := range volumes {
+		err = volume.PrepareVolumeForSnapshot(ctx, creds)
+		if err != nil {
+			errList = append(errList, err)
+		}
+	}
+	if len(errList) > 0 {
+		return nil, status.Errorf(
+			codes.Aborted,
+			"failed to prepare volumes for snapshot: %v",
+			errList)
+	}
```

Comment:
Dont we need to do this for the volumes if the groupSnapshot is already exists when we call `GetVolumeGroupSnapshotByName`?

Reply:
> Dont we need to do this for the volumes if the groupSnapshot is already exists when we call `GetVolumeGroupSnapshotByName`?

Those volumes would have already gone through this check in previous calls.